File: WePeiYang/lostfound/src/main/java/com/twtstudio/tjwhm/lostfound/waterfall/WaterfallContract.java
package com.twtstudio.tjwhm.lostfound.waterfall;

import com.twtstudio.tjwhm.lostfound.base.BaseContract;

/**
 * Created by tjwhm on 2017/7/2.
 **/

public interface WaterfallContract {
    public interface WaterfallView extends BaseContract.BaseView {
        void setWaterfallData(WaterfallBean waterfallBean);

        void loadWaterfallDataWithType(int type);
    }

    public interface WaterfallPresenter extends BaseContract.BasePresenter {
        void setWaterfallData(WaterfallBean waterfallBean);

        void loadWaterfallData(String lostOrFound, int page);

        void loadWaterfallDataWithType(String lostOrFound,int page,int type);
    }
}

File: WePeiYang/lostfound2/src/main/java/com/yookiely/lostfond2/waterfall/WaterfallContract.kt
package com.yookiely.lostfond2.waterfall

import com.yookiely.lostfond2.service.MyListDataOrSearchBean

interface WaterfallContract {

    interface WaterfallView {
        fun setWaterfallData(waterfallBean: List<MyListDataOrSearchBean>)

        fun loadWaterfallDataWithCondition(type: Int, time: Int)
    }

    interface WaterfallPresenter {
        fun setWaterfallData(waterfallBean: List<MyListDataOrSearchBean>)

        fun loadWaterfallData(lostOrFound: String, page: Int, time: Int)

        fun loadWaterfallDataWithCondition(lostOrFound: String, page: Int, type: Int, time: Int)
    }
}